### PR TITLE
Add API Endpoint for Listing LLM Provider Models

### PR DESCRIPTION
**Pull request by sajosam**
**Description:**

1. Implemented a new API to list available models for a given LLM provider.
2. This enhancement improves user access to LLM models by provider.

### CODE REVIEW

File: app/services/model_reader.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import requests
```
Comment (agberoz) [inline app/services/model_reader.py:1]: Do these in Corresponding LLM loader classes

**Review by agberoz — CHANGES_REQUESTED** [review]

File: app/loaders/togethor/loader.py [hunk 2]
```diff
@@ -89,4 +90,25 @@ def messages_format(self, prompt, previous_messages) -> list:
...
+
+    def get_models(self):
+        """
+        Retrieve models from the TogetherAI API.
+        Returns:
+            List of TogetherAI model names or an error message.
+        """
+        url = "https://api.together.xyz/v1/models"
+        headers = {
+            "Authorization": f"Bearer {self.model_config.api_key}",
```
Comment (agberoz) [inline app/loaders/togethor/loader.py:103]: Use self.model_config.get("api_key", "")

File: app/loaders/openai/loader.py [hunk 3]
```diff
@@ -86,4 +87,25 @@ def messages_format(self, prompt, previous_messages) -> list:
...
+    def get_models(self):
+        """
+        Retrieve models from the OpenAI API.
+        Returns:
+            List of OpenAI model names or an error message.
+        """
+        url = "https://api.openai.com/v1/models"
+        headers = {
+            "Authorization": f"Bearer {self.model_config.api_key}",
+        }
```
Comment (agberoz) [inline app/loaders/openai/loader.py:101]: Use self.model_config.get("api_key", "")

**Review by agberoz — CHANGES_REQUESTED** [review]

**Comment by agberoz** [comment]
@sajosam , Attach postman screen shot of API response

**Comment by sajosam** [comment]
@agberoz Here is the reponse,
image

**Review by agberoz — DISMISSED** [review]
LGTM

File: app/api/v1/connector.py [hunk 4]
```diff
@@ -87,7 +87,7 @@ def get_connector(connector_id: int, db: Session = Depends(get_db)):
 async def upload_document_datsource(
     file: UploadFile = File(...)
 ):
-
```
Comment (ashmilhussain) [inline app/api/v1/connector.py:90]: unwanted line changes

**Review by ashmilhussain — COMMENTED** [review]
@sajosam  check for unrelevent line and space additions

**Comment by sajosam** [comment]
@ashmilhussain  The spacing changes were auto-corrected by the pre-commit linter.

**Pull request merged**